src/routes/customer_product.js

```diff
@@ -97,4 +100,51 @@ productRouter.get('/others', async(req,res)=>{
     const items = await display.length
     res.render("./customers/page-listing-grid",{display,items});
 })
+
+// Edited by JJ
+productRouter.get('/detail/:id', async(req,res)=>{
+    try{
+        const productdetail = await product.findByPk(req.params.id)
+        const review = await Pevaluation.findAll({
+            where: {ProductId: req.params.id},
+            include: User
+        })
+        const fivestar = []
+        const fourstar = []
+        const threestar = []
+        const twostar = []
+        const onestar = []
+        for (let index = 0; index < review.length; index++) {
+            if (review[index]["ProductRating"]==5){
```

Comment:
Maybe bind `review[index]["ProductRating"]` into a variable
```suggestion
            const productRating = review[index]["ProductRating"];
            if (productRating == 5){
```